src/graph/graph.c

```diff
@@ -182,34 +182,31 @@ static inline size_t _Graph_EdgeCap(const Graph *g) {
 	return g->edges->itemCap;
 }
 
-// Locates edges connecting src to destination.
-void _Graph_GetEdgesConnectingNodes(const Graph *g, NodeID src, NodeID dest, int r, Edge **edges) {
-	ASSERT(g && src < Graph_RequiredMatrixDim(g) && dest < Graph_RequiredMatrixDim(g) &&
-		   r < Graph_RelationTypeCount(g));
-
+static void _CollectEdgesFromEntry
+(
+	const Graph *g,
+	NodeID src,
+	NodeID dest,
+	int r,
+	EdgeID edgeId,
+	Edge **edges
+) {
 	Edge e;
```

Comment:
Since the `Edge` struct has a lot of members that aren't explicitly set in this function, it would be safer to instantiate it with something like `Edge e = { 0 };`

Reply:
Good call